File: backend/app/models/conferencia.py
from sqlalchemy import Column, Integer, DateTime, Enum
from datetime import datetime

from app.database.connection import Base
from app.enums.conferencia_enums import StatusConferencia


class Conferencia(Base):
    __tablename__ = "conferencias"

    id = Column(Integer, primary_key=True, index=True)

    estabelecimento_id = Column(Integer)
    usuario_id = Column(Integer)

    quantidade_reaberturas = Column(Integer, default=0)
    versao = Column(Integer, default=1)

    status = Column(
        Enum(StatusConferencia),
        default=StatusConferencia.RASCUNHO
    )

    data_inicio = Column(
        DateTime,
          default=datetime.utcnow
    )

File: coletor_mobile/lib/models/conferencia.dart
class Conferencia {
  final int id;
  final int estabelecimentoId;
  final int usuarioId;
  final String status;
  final DateTime dataInicio;

  Conferencia({
    required this.id,
    required this.estabelecimentoId,
    required this.usuarioId,
    required this.status,
    required this.dataInicio,
  });

  factory Conferencia.fromJson(Map<String, dynamic> json) {
    return Conferencia(
      id: json['id'],
      estabelecimentoId: json['estabelecimento_id'],
      usuarioId: json['usuario_id'],
      status: json['status'],
      dataInicio: DateTime.parse(json['data_inicio']),
    );
  }
}
